test/integration/Main.hs

```diff
@@ -216,17 +219,27 @@ testScenarioSolution _ci _em =
         , testSolution Default "Testing/710-multi-robot"
         , testSolution Default "Testing/920-meet"
         , testSolution Default "Testing/955-heading"
+        , testSolution' Default "Testing/397-wrong-missing" AllowBadErrors $ \g -> do
+            let msgs =
+                  (g ^. messageQueue . to seqToTexts)
+                    <> (g ^.. robotMap . traverse . robotLog . to seqToTexts . traverse)
```

Comment:
Those are some impressive optics. :nerd_face:

Reply:
Haha, thanks, though I don't think it's that bad: `^..` means to list all the targets of a `Traversal` instead of just projecting out a single value.  `traverse` just gets all the sub-things.  So this means "get the robot map, then get all the robots out of it, and for each one get its log, convert it to a list of texts..." The last `traverse` is just so we get a single list of texts instead of a list of lists.